block_alerts.php

```diff
@@ -77,19 +77,45 @@ public function get_content(): stdClass {
         return $this->content;
     }
 
+    /**
+     * Return if user has archetype editingteacher.
+     *
+     */
+    public static function is_teacher(): bool {
+        global $DB, $USER;
+        // Get id's from role where archetype is editingteacher.
+        $roles = $DB->get_fieldset('role', 'id', ['archetype' => 'editingteacher']);
```

Comment:
Is it intended to include only the “editingteacher” archetype? What about other archetypes, such as:
	•	Manager: A system-level role used to manage courses without being directly enrolled in them.
	•	Course Creator: A system-level role used to create new courses.
	•	Teacher: A course-level role used to grade students but not add or edit activities.